pkg/processor/trigger/http/trigger.go

```diff
@@ -145,6 +157,35 @@ func (h *http) Stop(force bool) (functionconfig.Checkpoint, error) {
 	return nil, nil
 }
 
+func (h *http) StartBatcher(batchTimeout time.Duration, workerAvailabilityTimeout time.Duration) {
+	h.Logger.DebugWith("Start batcher", "batcherConfiguration", h.configuration.Batch)
+	for {
+		batch, responseChans := h.Batcher.WaitForBatch(batchTimeout)
+
+		h.Logger.Debug("Starting to process batch")
+		// allocate a worker
+		workerInstance, workerIndex, err := h.allocateWorker(workerAvailabilityTimeout)
+		if err != nil {
+			h.UpdateStatistics(false, uint64(len(batch)))
+			workerError := errors.Wrap(err, "Failed to allocate worker")
+			for _, channel := range responseChans {
+				go channel.Write(h.Logger, &runtime.ResponseWithErrors{SubmitError: workerError})
+			}
+		}
+		h.timeouts[workerIndex] = 0
+		h.answering[workerIndex] = 0
```

Comment:
I remember you mentioning some http trigger specific thing that is needed here with these `0` values. Probably worth adding a comment explaining this.